pkg/controllers/report/aggregate/controller.go

```diff
@@ -131,44 +183,81 @@ func NewController(
 	if _, _, err := controllerutils.AddDelayedDefaultEventHandlers(logger, cephrInformer.Informer(), c.frontQueue, enqueueDelay); err != nil {
 		logger.Error(err, "failed to register event handlers")
 	}
-	enqueueAll := func() {
-		selector := labels.SelectorFromSet(labels.Set{
-			kyverno.LabelAppManagedBy: kyverno.ValueKyvernoApp,
-		})
+	// enqueueReportsForPolicy queues only reports affected by a specific policy change using the cache.
+	enqueueReportsForPolicy := func(o metav1.Object) {
```

Comment:
Can we change it to a generic enqueue function to enqueue based on policy types?